golang/subscriber.go

```diff
@@ -0,0 +1,161 @@
+package golang
+
+import (
+	"context"
+	"sync"
+
+	v2 "github.com/cloudevents/sdk-go/v2"
+	"github.com/linkall-labs/vanus/observability/log"
+	proxypb "github.com/linkall-labs/vanus/proto/pkg/proxy"
+)
+
+type ackCallback func(result bool) error
```

Comment:
can use the enum to represent result?

Reply:
or is this necessary?

Reply:
optimize in a separate pr and remains unchanged in this pr.